backends/backends.c

```diff
@@ -658,7 +719,8 @@ void *backends_main(void *ptr) {
         size_t count_dims_total = 0;
 
 #if ENABLE_PROMETHEUS_REMOTE_WRITE
-        clear_write_request();
+        if(do_prometheus_remote_write)
```

Comment:
isn't a bit too much to have both a condition and the if-def that kills the code if prometheus not available?

Reply:
No, because ENABLE_PROMETHEUS_REMOTE_WRITE can be defined when the remote write backend is not enabled.

Reply:
I am not following, do_prometheus_remote_write=1 is set once at some point, with only obvious condition ENABLE_PROMETHEUS_REMOTE_WRITE to be set

Reply:
`ENABLE_PROMETHEUS_REMOTE_WRITE` is defined by autoconf if all required libraries are present in the system. `do_prometheus_remote_write` is set to 1 if `ENABLE_PROMETHEUS_REMOTE_WRITE` is defined AND the user enabled the prometheus remote write backend in the configuration.